utils/gen-styleguide-prod-config.js

```diff
@@ -0,0 +1,188 @@
+        // whole thing that reads the text and parses things out, but we don't
+        // need to. We'll just recreate the executed property here.
+        switch (entry[0]) {
+            case "webpackConfig":
+                lines.push("    webpackConfig: createConfig([babel(), postcss()]),");
+                break;
+
+            case "serverPort":
+                // Skip. We do a different port for prod so that prod and dev servers can coexist, if necessary.
+                break;
+
+            default:
+                lines.push(`    ${entry[0]}: ${JSON.stringify(entry[1])},`);
+                break;
+        }
+    });
+    // Let's specify the production server port.
+    lines.push(`    serverPort: 6061,`);
+    lines.push("};\n");
+
+    const data = lines.join("\n");
+    fs.writeFileSync(filepath, data, "utf8");
+}
+
+/**
+ * This function recurses through the sections and sub-sections, removing
+ * any that are marked as private. It returns the modified array with the
+ * private sections removed.
+ */
+function removePrivateSections(sections) {
+    if (!sections) {
+        return sections;
+    }
+    for (let i = sections.length; i > 0; i--) {
+        const sectionIndex = i - 1;
+        if (sections[sectionIndex].private) {
+            // eslint-disable-next-line no-console
+            console.log(`Removing private section ${sections[sectionIndex].name}`);
+
+            // If the section is private, we presume all content of the section is
+            // private too.
+            sections.splice(sectionIndex, 1);
+        } else {
+            // And now, we move into the chilren of the section and repeat.
+            sections[sectionIndex].sections =
+                removePrivateSections(sections[sectionIndex].sections);
+        }
+    }
+    return sections;
+}
+
+function maybeGetPackagePathForSection(section) {
+    const getPackageDirFromPath = filepath => {
+        if (!filepath) {
+            return null;
+        }
+
+        do {
+            const parentPath = path.dirname(filepath);
+            if (parentPath) {
+                const parentName = path.basename(parentPath);
+                if (parentName === "packages") {
+                    return filepath;
+                } else if (parentName === parentPath) {
+                    break;
+                } else {
+                    filepath = parentPath;
+                }
+            }
+        } while (filepath);
+
+        return null;
+    }
+
+    if (section.content) {
+        return getPackageDirFromPath(section.content);
+    } else {
+        // Other option is to get one of the component file paths and use that.
+        const files = getComponentFilesFromSection(section);
+        if (files.length) {
+            return getPackageDirFromPath(files[0]);
+        };
+    }
+
+    // We get here, we got nothing.
+    return null;
+}
+
+function getVersionInfoForSection(section) {
```

Comment:
Hmm - this gets a bit more than just the version info - would be good to clarify this (and add a comment explaining it).